Lombiq.VueJs/Services/VueSingleFileComponentShapeTemplateViewEngine.cs

```diff
@@ -20,54 +22,128 @@ public class VueSingleFileComponentShapeTemplateViewEngine : IShapeTemplateViewE
+        template = template
+            .GetParenthesisRanges("<!--", "-->")
+            .InvertRanges(template.Length)
+            .Join(template);
 
         var shapeName = displayContext.Value.Metadata.Type;
-        var stringLocalizer = _stringLocalizerFactory.Create("Vue.js SFC", shapeName);
+        var builder = new StringBuilder($"<script type=\"x-template\" class=\"{shapeName}\">");
 
-        foreach (var range in localizationRanges.OrderByDescending(range => range.End.Value))
+        var localizationRanges = template.GetParenthesisRanges("[[", "]]");
+        if (localizationRanges.Count > 0)
         {
-            var (before, expression, after) = template.Partition(range);
-            var text = expression[2..^2].Trim();
-            template = before + WebUtility.HtmlEncode(stringLocalizer[text]) + after;
+            await LocalizeRangesAsync(builder, template, localizationRanges, displayContext);
+        }
+        else
+        {
+            builder.Append(template);
         }
 
-        template = StringHelper.CreateInvariant($"<script type=\"x-template\" class=\"{shapeName}\">{template}</script>");
+        builder.Append("</script>");
 
         var entries = new List<object>();
         foreach (var amender in _amenders) entries.AddRange(await amender.PrependAsync(shapeName));
-        entries.Add(new HtmlString(template));
+        entries.Add(new HtmlString(builder.ToString()));
         foreach (var amender in _amenders) entries.AddRange(await amender.AppendAsync(shapeName));
 
         return new HtmlContentBuilder(entries);
     }
 
+    private async Task LocalizeRangesAsync(
+        StringBuilder builder,
+        string template,
+        IList<Range> localizationRanges,
+        DisplayContext context)
+    {
+        var shapeName = context.Value.Metadata.Type;
+        var stringLocalizerLazy = new Lazy<IStringLocalizer>(() => _stringLocalizerFactory.Create("Vue.js SFC", shapeName));
+        var htmlLocalizerLazy = new Lazy<IHtmlLocalizer>(() => _htmlLocalizerFactory.Create("Vue.js SFC HTML", shapeName));
+
+        var startIndex = new Index(0);
+        foreach (var range in localizationRanges)
+        {
+            // Insert content before this range.
+            builder.Append(template[startIndex..range.Start]);
+            startIndex = range.End;
+
+            var expression = template[range];
+            string html;
+
+            // Include a logger warning if the inner spacing is missing. This will cause failures e.g. during UI tests,
+            // and so ensure correct formatting.
+            if (expression[2] is not '{' and not ' ')
+            {
+                _logger.LogWarning(
+                    "Vue SFC localization strings should follow the following formats: [[ text ]], [[{{ html }}]] or " +
+                    "[[{{converter}} input ]]. Please include the inner spacing to ensure future compatibility. Your " +
```

Comment:
```suggestion
                    "Vue SFC localization strings should follow the following formats: [[ text ]], [[{ html }]] or " +
                    "[[{converter} input ]]. Please include the inner spacing to ensure future compatibility. Your " +
```
Isn't this the correct syntax?

Reply:
`{{` is an escape sequence, `[[{converter} input ]]` would tell the logger to look for an optional argument named "converter" See the `\"{Expression}\"` on line 111.

If you let this execute, the log will be like:
```
2024-04-29 18:48:35.6921|Default|00-e95b88e0bb813a610f82e14827b7410c-ea9a812249d69f0a-00||Lombiq.VueJs.Services.VueSingleFileComponentShapeTemplateViewEngine|WARN|Vue SFC localization strings should follow the following formats: [[ text ]], [[{ html }]] or [[{converter} input ]]. Please include the inner spacing to ensure future compatibility. Your expression was: "[[Test free of charge ]]". 
```